src/Swashbuckle.AspNetCore.Filters/Examples/MvcOutputFormatter.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Net.Http.Headers;
+using System.Linq;
+
+namespace Swashbuckle.AspNetCore.Filters
+{
+    internal class MvcOutputFormatter
+    {
+        private readonly object initializeLock = new object();
+        private bool initializedOutputFormatterSelector;
+
+        private readonly IOptions<MvcOptions> selectorOptions;
+        private readonly ILoggerFactory loggerFactory;
+
+        private OutputFormatterSelector outputFormatterSelector;
+        private OutputFormatterSelector OutputFormatterSelector
+        {
+            get
+            {
+                lock (initializeLock)
+                {
+                    if (!initializedOutputFormatterSelector)
+                    {
+                        if (selectorOptions != null)
+                        {
+                            outputFormatterSelector = new DefaultOutputFormatterSelector(
+                                selectorOptions,
+                                loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory)));
+                        }
+                        initializedOutputFormatterSelector = true;
+                    }
+                }
+
+                return outputFormatterSelector;
+            }
+        }
+
+        public MvcOutputFormatter(IOptions<MvcOptions> options, ILoggerFactory loggerFactory)
+        {
+            this.initializedOutputFormatterSelector = false;
+            this.selectorOptions = GetSelectorOptions(options);
+            this.loggerFactory = loggerFactory;
+        }
+
+        private static IOptions<MvcOptions> GetSelectorOptions(IOptions<MvcOptions> options)
+        {
+            if (options?.Value?.OutputFormatters == null || !options.Value.OutputFormatters.Any())
+            {
+                return null;
+            }
+
+            var selectorOptions = new MvcOptions
+            {
+                ReturnHttpNotAcceptable = true,

```

Comment:
Just wondering why these options are being set

Reply:
Without these, MVC defaults to returning JSON if it does not find a formatter for XML, which is not what we want :)